contracts/SDRewardManager.sol

```diff
@@ -0,0 +1,124 @@
+pragma solidity 0.8.16;
+
+import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import { SafeERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import { IStaderConfig } from "./interfaces/IStaderConfig.sol";
+import { UtilLib } from "./library/UtilLib.sol";
+
+/**
+ * @title SDRewardManager
+ * @notice This contract is responsible to add SD rewards to the socializing pool
+ */
+contract SDRewardManager is Initializable {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    struct SDRewardEntry {
+        uint256 cycleNumber;
+        uint256 amount;
+        bool approved;
+    }
+
+    IStaderConfig public staderConfig;
+
+    uint256 public latestCycleNumber;
+
+    // Mapping of cycle numbers to reward entries
+    mapping(uint256 => SDRewardEntry) public rewardEntries;
+
+    // Event emitted when a new reward entry is created
+    event NewRewardEntry(uint256 indexed cycleNumber, uint256 amount);
+
+    // Event emitted when a reward entry is approved
+    event RewardEntryApproved(uint256 indexed cycleNumber, uint256 amount);
+
+    error AccessDenied(address account);
+    error EntryNotFound(uint256 cycleNumber);
+    error EntryAlreadyRegistered(uint256 cycleNumber);
+    error EntryAlreadyApproved(uint256 cycleNumber);
+    error InvalidCycleNumber(uint256 cycleNumber);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /**
+     * @notice Initializes the contract with a Stader configuration address
+     * @param _staderConfig Address of the StaderConfig contract
+     */
+    function initialize(address _staderConfig) external initializer {
+        UtilLib.checkNonZeroAddress(_staderConfig);
+        staderConfig = IStaderConfig(_staderConfig);
+    }
+
+    /**
+     * @notice Adds a new reward entry for a specified cycle
+     * @param _cycleNumber The cycle number for the reward entry
+     * @param _amount The amount of SD to be rewarded
+     */
+    function addRewardEntry(uint256 _cycleNumber, uint256 _amount) external {
```

Comment:
if allowed user happens to pass a cycle_number bigger than required cycle_number, sdRewardManager would be blocked.

I would suggest to allow user to send `cycleNumber` instead read the `cycleNumber` from `socialisingPool`/`staderOracle`. this will also result in getting rid of few extra `cycleNumber` checks.

similarly in approveEntry, only the current cycle should be read from socialisingPool

Reply:
[837cf25](https://github.com/stader-labs/ethx/pull/253/commits/837cf25cc2c7604af369769a6963c61ffee61323)